src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,37 +1,107 @@
-import { useState } from 'react';
+import React, { useState, FC, FormEvent } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+interface NewMovieProps {
+  moviesFromServer: Movie[];
+  onAdd: (newMovie: Movie) => void;
+}
+
+export const NewMovie: FC<NewMovieProps> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [newMovie, setNewMovie] = useState<Movie>({
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  });
+
+  const requiredFields = [
+    newMovie.title.trim(),
+    newMovie.imgUrl.trim(),
+    newMovie.imdbUrl.trim(),
+    newMovie.imdbId.trim(),
+  ];
+
+  const isButtonDisabled = !requiredFields.every(Boolean);
+
+  const handleInputChange = (name: keyof Movie, value: string) => {
+    setNewMovie(prevMovie => ({
+      ...prevMovie,
+      [name]: value,
+    }));
+  };
```

Comment:
```suggestion
  const handleInputChange = (event) => {
    const { name, value } = event.target;
    
    setNewMovie(prevMovie => ({
      ...prevMovie,
      [name]: value,
    }));
  };
```
👍
You can even simplify and get name and value from the change event